tests/chainer_tests/training_tests/triggers_tests/test_early_stopping_trigger.py

```diff
@@ -0,0 +1,111 @@
+import unittest
+
+import chainer
+import numpy
+
+from chainer import testing
+from chainer import training
+from chainer.training import triggers
+from chainer.training import util
+
+
+class DummyUpdater(training.Updater):
+
+    def __init__(self):
+        self.iteration = 0
+        self.epoch = 0
+
+    def finalize(self):
+        pass
+
+    def get_all_optimizers(self):
+        return {}
+
+    def update(self):
+        self.iteration += 1
+        self.epoch += 1
+
+    def epoch(self):
+        return self.epoch
+
+    @property
+    def is_new_epoch(self):
+        return False
+
+    @property
+    def epoch_detail(self):
+        return self.epoch
+
+
+def _test_trigger(self, trigger, key, accuracies, expected):
+    updater = DummyUpdater()
+    trainer = training.Trainer(updater)
```

Comment:
Consider using `testing.get_trainer_with_mock_updater()` instead of defining your own mock updater (see https://github.com/chainer/chainer/blob/master/chainer/testing/training.py#L29).

Reply:
Updated to use `get_trainer_with_mock_updater`.